src/inventories/quantities/InventoryItemQuantitiesService.ts

```diff
@@ -0,0 +1,93 @@
+import RestService from '../../service/RestService';
+import RequestClientWrapper from '../../RequestClientWrapper';
+import Adjustment from './model/Adjustment';
+import { AxiosRequestConfig } from 'axios';
+import AdjustmentRequest from './model/AdjustmentRequest';
+import * as Q from 'q';
+import Resource from '../../model/Resource';
+import Version from '../../model/Version';
+import SubResource from '../../model/SubResource';
+import * as uuidV4 from 'uuid/v4';
+import InventoryItemQuantitiesResponse from './model/InventoryItemQuantitiesResponse';
+import InventoryItemQuantity from './model/InventoryItemQuantity';
+import RequestCallbackParams from '../../../src/model/RequestCallbackParams';
+import GenerateApiError from './../../utils/GenerateApiError';
+
+export default class InventoryQuantitiesService extends RestService {
+  EXPECTED_POST_STATUS: number = 201;
+  EXPECTED_GET_STATUS: number = 200;
+
+  constructor(private readonly client: RequestClientWrapper) {
+    super();
+  }
+
+  public adjust(adjustmentRequest: AdjustmentRequest): Promise<Adjustment> {
+    return this.createQuantityAdjustment(adjustmentRequest, SubResource.ADJUSTS);
+  }
+
+  public set(adjustmentRequest: AdjustmentRequest): Promise<Adjustment> {
+    return this.createQuantityAdjustment(adjustmentRequest, SubResource.SETS);
+  }
+
+  public retrieve(inventoryItemId: string): Promise<InventoryItemQuantity[]> {
+    return this.retrieveInventoryItemQuantities(inventoryItemId);
+  }
+
+  private retrieveInventoryItemQuantities(inventoryItemId: string)
+    : Promise<InventoryItemQuantity[]> {
+    return new Promise((resolve) => {
+      const requestUrl
+        = `/${Version.V1}${Resource.INVENTORIES}/${inventoryItemId}/${SubResource.QUANTITIES}`;
+      this.client.get(requestUrl, {}, (error, response, body) => {
+        const requestResponse: RequestCallbackParams = {
+          error,
+          response,
+          body
+        };
+        resolve(this.mapInventoryItemsQuantitiesPromise(requestUrl, requestResponse, this.EXPECTED_GET_STATUS));
+      });
+    });
+  }
+
+  private createQuantityAdjustment(adjustmentRequest: AdjustmentRequest, operation: string) {
+    const deferred = Q.defer<Adjustment>();
+    const inventoryItemId = adjustmentRequest.inventoryItemId;
+    const requestUrl
+      = `/${Version.V1}${Resource.INVENTORIES}/${inventoryItemId}/${SubResource.QUANTITIES}/${operation}`;
+    let idempotentKey = uuidV4();
+    if (adjustmentRequest.idempotentKey) {
+      idempotentKey = adjustmentRequest.idempotentKey;
+    }
+    const options: AxiosRequestConfig = {
+      data: adjustmentRequest,
+      headers: {
+        'X-Channel-Ape-Idempotent-Key': idempotentKey

```

Comment:
@mb-ca  we could do that, but I don't believe it makes sense now since it's only on this request right. Also, we want to allow the client to be able to use their own idempotentKey. Hence the null check in this method. This would add another optional method argument bumping that up to 5. 
If you really think we should future proof now, I'm thinking that we can make another method called this.client.idempotentPost(...) with the optional parameter to provide their own idempotent Key but I think that is a little overkill for only the 1 request.

Please let me know what you think.

Reply:
@ryankazokas Okay good point, this is fine for now then.